frontend/src/metabase/visualizations/visualizations/PieChart.jsx

```diff
@@ -195,13 +195,15 @@ export default class PieChart extends Component {
   };
 
   componentDidUpdate() {
-    const groupElement = ReactDOM.findDOMNode(this.refs.group);
-    const detailElement = ReactDOM.findDOMNode(this.refs.detail);
-    if (groupElement.getBoundingClientRect().width < 120) {
-      detailElement.classList.add("hide");
-    } else {
-      detailElement.classList.remove("hide");
-    }
+    requestAnimationFrame(() => {
+      const groupElement = ReactDOM.findDOMNode(this.refs.group);
+      const detailElement = ReactDOM.findDOMNode(this.refs.detail);
```

Comment:
This is something I'm working on in https://github.com/metabase/metabase/pull/15098. Our code base is riddled with similar examples. 

Not only do we use string refs `ref="group"`, but I also don't understand the logic of using `ReactDOM.findDOMNode` on the `ref`. Kind of defeats the purpose for using refs in the first place, right?

Anyway, I know this was not the point of this request but maybe it would be a good start to discontinue the practice of using string refs.